envs/quadrotors/quad_multi_model.py

```diff
@@ -33,27 +34,50 @@ def __init__(self, cfg, obs_space, timing, self_obs_dim=18, neighbor_obs_dim=6,
             nonlinearity(cfg)
         )
 
+        # encode the obstacle observations
+        if self.obstacle_mode != 'None':
+            self.obstacle_obs_dim = obstacle_obs_dim
+            self.obstacle_hidden_size = obstacle_hidden_size
+            self.obstacle_encoder = nn.Sequential(
+                fc_layer(self.obstacle_obs_dim, self.obstacle_hidden_size, spec_norm=self.use_spectral_norm),
+                nonlinearity(cfg),
+                fc_layer(self.obstacle_hidden_size, self.obstacle_hidden_size, spec_norm=self.use_spectral_norm),
+                nonlinearity(cfg),
+            )
+            self.obstacle_encoder_out_size = calc_num_elements(self.obstacle_encoder, (self.obstacle_obs_dim,))
+
         self.self_encoder_out_size = calc_num_elements(self.self_encoder, (self.self_obs_dim,))
         self.neighbor_encoder_out_size = calc_num_elements(self.neighbor_encoder, (self.neighbor_obs_dim,))
 
-        # Feed forward self obs and neighbor obs after concatenation
-        self.feed_forward = fc_layer(self.self_encoder_out_size + self.neighbor_encoder_out_size, cfg.hidden_size, spec_norm=self.use_spectral_norm)
-
+        if self.obstacle_mode == 'None':
+            # Feed forward self obs and neighbor obs after concatenation
+            self.feed_forward = fc_layer(self.self_encoder_out_size + self.neighbor_encoder_out_size, cfg.hidden_size, spec_norm=self.use_spectral_norm)
+        else:
+            self.feed_forward = fc_layer(self.self_encoder_out_size + self.neighbor_encoder_out_size + self.obstacle_encoder_out_size, cfg.hidden_size, spec_norm=self.use_spectral_norm)
```

Comment:
Lots of code duplication here.
Just calculate the total encoder output size as you go and use it to create fc_layer. Please try to minimize code duplication as much as you can